prospero-common/src/test/java/org/wildfly/prospero/model/ProvisioningConfigTest.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.prospero.model;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+
+
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.Assert.*;
+
+public class ProvisioningConfigTest {
+
+    private final ProvisioningConfig provisioningConfig = new ProvisioningConfig(new ArrayList<>(), new ArrayList<>());
+
+    @Test
+    public void addRepositoryIgnoresChangesIfExistingRepo() throws Exception {
+        provisioningConfig.addRepository(new RepositoryRef("existing", "file:///foo.bar"));
+
+        assertFalse(provisioningConfig.addRepository(new RepositoryRef("existing", "file:///foo.bar")));
+
+        assertThat(provisioningConfig.getRepositories()).containsExactly(
+                new RepositoryRef("existing", "file:///foo.bar")
+        );
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void addRepositoryThrowsErrorIfSameIdDifferentUrl() throws Exception {
+        provisioningConfig.addRepository(new RepositoryRef("existing", "file:///foo.bar"));
+
+        assertFalse(provisioningConfig.addRepository(new RepositoryRef("existing", "file:///different.url")));
+
+        assertThat(provisioningConfig.getRepositories()).containsExactly(
+                new RepositoryRef("existing", "file:///foo.bar")
+        );
```

Comment:
I think these following assert statements would never be executed, as the exception would be thrown before them.

Reply:
Yes that's true, forgot to remove those